WcaOnRails/app/controllers/api/v0/api_controller.rb

```diff
@@ -9,9 +10,13 @@ def bad_api_parameter(e)
 
   DEFAULT_API_RESULT_LIMIT = 20
 
+  def current_resource_owner
```

Comment:
`current_api_user` sounds good to me.

That is true about caching falsey values (it's also seems very likely that people will try to use our API without logging in, although maybe we should not allow that). http://stackoverflow.com/questions/11158972/how-can-i-memoize-a-method-that-may-return-true-false-or-nil-in-ruby suggests the following rewrite:

```ruby
def current_api_user
  @current_api_user if defined?(@current_api_user)
  @current_api_user = doorkeeper_token ? User.find_by_id(doorkeeper_token.resource_owner_id) : nil
end
```